### PR TITLE
Visualize ocean floor geo data in Superset (with Trino, from S3)

### PR DESCRIPTION
**Pull request by fhennig**
Working on this: https://github.com/stackabletech/marispace-x/issues/71

Data uploaded here: https://repo.stackable.tech/repository/misc/marispace/multibeam_data_point_density_example.parquet

The coordinates use this reference frame: https://epsg.io/32632

----

The PR also makes a change in the `superset-trino-s3` Stack which is also used by the Taxi demo. The change raises the row limit in superset, so larger queries can be made. This is necessary for this demo, to get a complete picture of the ocean floor.

### CODE REVIEW

**Comment by fhennig** [comment]
It was straightforward to adapt the taxi data demo, but now I can't find a good Chart type to plot the data.

The data would best be plotted with a surface plot, but Superset doesn't offer this.

Maybe a view of the data can be created with already regular buckets and then the heatmap type can be used.

**Comment by fhennig** [comment]
image

This is the visualization I have now. I got the asset working too.

Now I will just clean up some file naming.

**Comment by fhennig** [comment]
before merging, `046a0e5c6589644c790bbeba74406c8d18a1e4ff` needs to be replaced with `main` in the file references.

File: demos/demos-v2.yaml [hunk 1]
```diff
@@ -147,6 +147,25 @@ demos:
       cpu: 6800m
       memory: 15822Mi
       pvc: 28Gi
+  trino-subsea-data:
+    description: TODO
+    documentation: TODO
```
Comment (sbernauer) [inline demos/demos-v2.yaml:152]: Reminder to add description and docs :)

File: demos/demos-v2.yaml [hunk 2]
```diff
@@ -147,6 +147,25 @@ demos:
...
+    labels:
+      - trino
+      - superset
+      - minio
+      - s3
+      - parquet
+    manifests:
+      - plainYaml: https://raw.githubusercontent.com/stackabletech/demos/feat/marispace-demo/demos/trino-subsea-data/load-test-data.yaml
+      - plainYaml: https://raw.githubusercontent.com/stackabletech/demos/feat/marispace-demo/demos/trino-subsea-data/create-table-in-trino.yaml
+      - plainYaml: https://raw.githubusercontent.com/stackabletech/demos/feat/marispace-demo/demos/trino-subsea-data/setup-superset.yaml
```
Comment (sbernauer) [inline demos/demos-v2.yaml:163]: Reminder to change links

File: demos/trino-subsea-data/create-table-in-trino.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    run_query(connection, "CREATE SCHEMA IF NOT EXISTS hive.demo WITH (location = 's3a://demo/')")
+    run_query(connection, """
+        CREATE TABLE IF NOT EXISTS hive.demo.marispace (
+            footprint_x DOUBLE,
+            footprint_y DOUBLE,
+            water_depth DOUBLE,
+            data_point_density DOUBLE,
+            geometry VARBINARY
+        ) WITH (
+            external_location = 's3a://demo/',
```
Comment (sbernauer) [inline demos/trino-subsea-data/create-table-in-trino.yaml:69]: I would suggest to have the same folders in S3 as in Trino for consistency - we also do this in other demos.
Remember to update `demos/trino-subsea-data/load-test-data.yaml` as well
```suggestion
            external_location = 's3a://demo/marispace',
```

File: stacks/stacks-v2.yaml [hunk 4]
```diff
@@ -303,7 +303,7 @@ stacks:
       - plainYaml: https://raw.githubusercontent.com/stackabletech/demos/main/stacks/trino-superset-s3/s3-connection.yaml
       - plainYaml: https://raw.githubusercontent.com/stackabletech/demos/main/stacks/trino-superset-s3/hive-metastore.yaml
       - plainYaml: https://raw.githubusercontent.com/stackabletech/demos/main/stacks/trino-superset-s3/trino.yaml
-      - plainYaml: https://raw.githubusercontent.com/stackabletech/demos/main/stacks/trino-superset-s3/superset.yaml
+      - plainYaml: https://raw.githubusercontent.com/stackabletech/demos/feat/marispace-demo/stacks/trino-superset-s3/superset.yaml
```
Comment (sbernauer) [inline stacks/stacks-v2.yaml:306]: Reminder to change back

File: stacks/trino-superset-s3/superset.yaml [hunk 5]
```diff
@@ -14,6 +14,10 @@ spec:
     roleGroups:
       default:
         replicas: 1
+        configOverrides:
+          superset_config.py:
+            ROW_LIMIT: "200000"
+            SQL_MAX_ROW: "200000"
```
Comment (sbernauer) [inline stacks/trino-superset-s3/superset.yaml:20]:
```suggestion
          superset_config.py:
            # Needed by trino-subsea-data demo
            ROW_LIMIT: "200000"
            SQL_MAX_ROW: "200000"
```

**Review by sbernauer — CHANGES_REQUESTED** [review]
LGTM in general, but did not try it out yet

File: demos/demos-v2.yaml (hunk 1 above)
Comment (fhennig) [reply demos/demos-v2.yaml:152]: I'll ask @stefanigel if we want to put it in the docs too or what. Will bring it up tomorrow in our Gaia-X Meeting.

**Comment by fhennig** [comment]
I have now attempted for multiple hours to get the data visualized in the `deck.gl Heatmap` plot type, but I can't get the coordinate conversion right. 

What I did: use calculated columns in Superset to calculate lat/lon coordinates
- `lat = footprint_y / 111059.585476`
- `lon = footprint_x / 111059.585476 + 6`

This is based on the fixed distance that 1 degree does for lat, and then the 6 degree offset for the EPSG:32632 reference frame in eastern direction. But the location I get is still off:

image

(The big circle is where we get, the smaller one where we should be).

You can use [this tool](https://www.koordinaten-umrechner.de/decimal/54.042291,14.041950?karte=OpenStreetMap&zoom=10) to calculate the actual coordinates that you should get.

I have asked north.io for more help on concrete conversion numbers to use, but they couldn't readily provide any.

File: demos/trino-subsea-data/create-table-in-trino.yaml (hunk 3 above)
Comment (fhennig) [reply demos/trino-subsea-data/create-table-in-trino.yaml]: https://github.com/stackabletech/demos/pull/88/commits/54d0d07248a6af38373ced722a97a1b2ca146a94

File: demos/demos-v2.yaml (hunk 1 above)
Comment (sbernauer) [reply demos/demos-v2.yaml]: works for me

Comment (sbernauer) [reply demos/demos-v2.yaml]: Please remove the documentation field, it's optional ;)
https://github.com/stackabletech/stackable-cockpit/blob/3e2ed32c3ce0303c18463957c45012ef92419240/rust/stackable-cockpit/src/platform/demo/spec.rs#L64-L66

File: stacks/stacks-v2.yaml (hunk 4 above)
Comment (fhennig) [reply stacks/stacks-v2.yaml]: https://github.com/stackabletech/demos/pull/88/commits/2ece0adffa9f874b5eed5654827c1d478bab356b

File: demos/demos-v2.yaml (hunk 2 above)
Comment (fhennig) [reply demos/demos-v2.yaml]: https://github.com/stackabletech/demos/pull/88/commits/2ece0adffa9f874b5eed5654827c1d478bab356b

File: demos/trino-subsea-data/setup-superset.yaml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+kind: Job
+metadata:
+  name: setup-superset
+spec:
+  template:
+    spec:
+      containers:
+        - name: setup-superset
+          image: docker.stackable.tech/stackable/testing-tools:0.2.0-stackable24.7.0
+          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/demos/feat/marispace-demo/demos/trino-subsea-data/superset-assets.zip && python -u /tmp/script/script.py"]
```
Comment (sbernauer) [inline demos/trino-subsea-data/setup-superset.yaml:12]:
```suggestion
          command: ["bash", "-c", "curl -o superset-assets.zip https://raw.githubusercontent.com/stackabletech/demos/main/demos/trino-subsea-data/superset-assets.zip && python -u /tmp/script/script.py"]
```

**Review by sbernauer — CHANGES_REQUESTED** [review]
LGTM otherwise

**Review by sbernauer — APPROVED** [review]

**Pull request merged**